### ISSUE TITLE
phpMyFAQ\User\TwoFactor::getSecret() after update from 3.1.12 to 3.2.0RC2

### ISSUE THREAD

**Issue by albertcd** [thorsten/phpMyFAQ#2517]
**Describe the bug**
Error when try to access to user control panel after update 3.2.0. RC2

**To Reproduce**
After update from 3.1.12 to 3.2.0 RC2 and going to "User  Control Panel" (https://domain.com/?action=ucp) show this fatal error:

```

phpMyFAQ Fatal error
Uncaught exception: 'TypeError'

Message: 'phpMyFAQ\User\TwoFactor::getSecret(): Return value must be of type string, null returned'

Stack trace:

#0 /var/www/vhosts/domain.com/kb.domain.com/ucp.php(52): phpMyFAQ\User\TwoFactor->getSecret()
#1 /var/www/vhosts/domain.com/kb.domain.com/index.php(793): require('...')
#2 {main}
Thrown in '/var/www/vhosts/domain.com/kb.domain.com/src/phpMyFAQ/User/TwoFactor.php' on line 64
```

**phpMyFAQ (please complete the following information):**

- phpMyFAQ version 3.2.0 RC2
- PHP version 8.1.20
- Database MariaDB 10.3.38
- Elasticsearch yes/no NO

**Comment by modelrailroader** [comment]
I just saw that this issue concerns "my" function and checked it:
Probably, you just have to change line 54 in ucp.php to 
` if (is_null($secret)) {`

This is because the default value for secret in the db is set to null in the update script. @thorsten 
But I can't test it currently...

**Comment by albertcd** [comment]
Today, after logout, i cannot login anymore.
I tried to reset password, and with new password neither.

I tried change the line
`try {`
to
`if (is_null($secret)) {`

but cannot login to phpMyFAQ.

Thanks u

**Comment by modelrailroader** [comment]
@albertcd Did you enabled 2FA? What is shown if you log in to your phpMyFAQ instance? Any error messages, maybe if you turn on DEBUG?

**Comment by modelrailroader** [comment]
> Today, after logout, i cannot login anymore. I tried to reset password, and with new password neither.
> 
> I tried change the line `try {` to `if (is_null($secret)) {`
> 
> but cannot login to phpMyFAQ.
> 
> Thanks u

That's wrong, it's my fault, I'm sorry. With your version of the database, you have to change line 53 in ucp.php to the line above, so:
`if ($secret === '') {` to ` if (is_null($secret)) {`

This should fix viewing ucp.php.
But this can't be the solution why you can't log in, so please answer my last questions too

**Comment by albertcd** [comment]
> > Today, after logout, i cannot login anymore. I tried to reset password, and with new password neither.
> > I tried change the line `try {` to `if (is_null($secret)) {`
> > but cannot login to phpMyFAQ.
> > Thanks u
> 
> That's wrong, it's my fault, I'm sorry. With your version of the database, you have to change line 53 in ucp.php to the line above, so: `if ($secret === '') {` to ` if (is_null($secret)) {`
> 
> This should fix viewing ucp.php. But this can't be the solution why you can't log in, so please answer my last questions too

I changed the line 52, now i have this lines:

```
$tfa = new TwoFactor($faqConfig);
$secret = $tfa->getSecret(CurrentUser::getFromSession($faqConfig));
if (is_null($secret)) {
     try {
            $secret = $tfa->generateSecret();
```

Now i can logging into phpMyFAQ (i don't understand why now is working, and this morning say "Bad user/password") and when i access to ucp, i get this error:

```
Uncaught exception: 'TypeError'

Message: 'phpMyFAQ\User\TwoFactor::getSecret(): Return value must be of type string, null returned'

Stack trace:

#0 /var/www/vhosts/domain.com/kb.domain.com/ucp.php(52): phpMyFAQ\User\TwoFactor->getSecret()
#1 /var/www/vhosts/domain.com/kb.domain.com/index.php(793): require('...')
#2 {main}
Thrown in '/var/www/vhosts/domain.com/kb.domain.com/src/phpMyFAQ/User/TwoFactor.php' on line 64
```

**Comment by albertcd** [comment]
I solved the access to ucp.php changing the line 64 of src/phpMyFAQ/User/TwoFactor.php to:

` return is_null($user->getUserData('secret')) ? '' : $user->getUserData('secret');`

**Comment by modelrailroader** [comment]
Yeah, you can do it like that, but alternatively, you should be able to change this line 
`public function getSecret(CurrentUser $user): string`
to
`public function getSecret(CurrentUser $user): string|null` 
in Twofactor.php as well.

**Comment by thorsten** [comment]
Fixed with #2521
